Mathlib/LinearAlgebra/FreeModule/Rank.lean

```diff
@@ -123,3 +137,109 @@ theorem rank_tensorProduct' (N : Type v) [AddCommGroup N] [Module R N] [Module.F
+  clear_value c s
+  simp_rw [← rep_e, Finset.sum_apply, Pi.smul_apply, Finset.smul_sum]
+  rw [Finset.sum_comm]
+  refine Finset.sum_eq_zero fun i hi ↦ ?_
+  replace eq0 := congr_arg L.subtype (congr_fun eq0 ⟨i, hi⟩)
+  rw [Finset.sum_apply, map_sum] at eq0
+  have : SMulCommClass Lᵐᵒᵖ K K := ⟨fun _ _ _ ↦ mul_assoc _ _ _⟩
+  simp_rw [smul_comm _ (c i), ← Finset.smul_sum]
+  erw [eq0, smul_zero]
+
+variable {K}
+
+open Function in
+theorem rank_fun_infinite {ι : Type v} [hι : Infinite ι] : Module.rank K (ι → K) = #(ι → K) := by
+  obtain ⟨⟨ιK, bK⟩⟩ := Module.Free.exists_basis (R := K) (M := ι → K)
+  obtain ⟨e⟩ := lift_mk_le'.mp ((aleph0_le_mk_iff.mpr hι).trans_eq (lift_uzero #ι).symm)
+  have := LinearMap.lift_rank_le_of_injective _ <|
+    LinearMap.funLeft_injective_of_surjective K K _ (invFun_surjective e.injective)
+  rw [lift_umax.{u,v}, lift_id'.{u,v}] at this
+  have key := (lift_le.{v}.mpr <| max_aleph0_card_le_rank_fun_nat K).trans this
+  rw [lift_max, lift_aleph0, max_le_iff] at key
+  haveI : Infinite ιK := by
+    rw [← aleph0_le_mk_iff, ← rank_eq_cardinal_basis' bK]; exact key.1
+  rw [bK.repr.toEquiv.cardinal_eq, mk_finsupp_lift_of_infinite,
+      lift_umax.{u,v}, lift_id'.{u,v}, ← rank_eq_cardinal_basis' bK, eq_comm, max_eq_left]
+  exact key.2
+
+/-- The **Erdős-Kaplansky Theorem**: the dual of an infinite-dimensional vector space
+  over a division ring has dimension equal to its cardinality. -/
+theorem rank_dual_eq_card_dual_of_aleph0_le_rank' {V : Type*} [AddCommGroup V] [Module K V]
+    (h : ℵ₀ ≤ Module.rank K V) : Module.rank Kᵐᵒᵖ (V →ₗ[K] K) = #(V →ₗ[K] K) := by
+  obtain ⟨⟨ι, b⟩⟩ := Module.Free.exists_basis (R := K) (M := V)
+  rw [rank_eq_cardinal_basis' b, aleph0_le_mk_iff] at h
+  have e := (b.constr Kᵐᵒᵖ (M' := K)).symm.trans
+    (LinearEquiv.piCongrRight fun _ ↦ MulOpposite.opLinearEquiv Kᵐᵒᵖ)
+  rw [e.rank_eq, e.toEquiv.cardinal_eq]
+  apply rank_fun_infinite
+
+/-- The **Erdős-Kaplansky Theorem** over a field. -/
+theorem rank_dual_eq_card_dual_of_aleph0_le_rank {K V} [Field K] [AddCommGroup V] [Module K V]
+    (h : ℵ₀ ≤ Module.rank K V) : Module.rank K (V →ₗ[K] K) = #(V →ₗ[K] K) := by
+  obtain ⟨⟨ι, b⟩⟩ := Module.Free.exists_basis (R := K) (M := V)
+  rw [rank_eq_cardinal_basis' b, aleph0_le_mk_iff] at h
+  have e := (b.constr K (M' := K)).symm
+  rw [e.rank_eq, e.toEquiv.cardinal_eq]
+  apply rank_fun_infinite
+
+theorem lift_rank_lt_rank_dual' {V : Type v} [AddCommGroup V] [Module K V]
+    (h : ℵ₀ ≤ Module.rank K V) :
+    Cardinal.lift.{u} (Module.rank K V) < Module.rank Kᵐᵒᵖ (V →ₗ[K] K) := by
+  obtain ⟨⟨ι, b⟩⟩ := Module.Free.exists_basis (R := K) (M := V)
+  rw [rank_eq_cardinal_basis' b, rank_dual_eq_card_dual_of_aleph0_le_rank' h,
+      ← (b.constr ℕ (M' := K)).toEquiv.cardinal_eq, mk_arrow]
+  apply cantor'
+  erw [nat_lt_lift_iff, one_lt_iff_nontrivial]
+  infer_instance
+
+theorem lift_rank_lt_rank_dual {K : Type u} {V : Type v} [Field K] [AddCommGroup V] [Module K V]
```

Comment:
Should we have versions of these last two results in the equi-universic setting, that avoid the `Cardinal.lift`?
Maybe they should have a `'` in their name, and the results should come with docstrings that cross-ref each other.

Reply:
Certainly! No need of `'`, just remove `lift` from the names. (I use `'` for the division ring version and no `'` for the field version in this PR.)